.github/workflows/website-test.yml

```diff
@@ -0,0 +1,81 @@
+name: "Website Tests"
+
+on:
+    pull_request:
+        types:
+            - opened
+            - reopened
+            - synchronize
+    push:
+        branches:
+            - master
+            - "v[0-9]+.[0-9]+"
+        tags:
+            - "(dev-)?v[0-9]+.[0-9]+.[0-9]+"
+
+jobs:
+    define-versions:
+        runs-on: ubuntu-latest
+        outputs:
+            fdiVersions: ${{ steps.versions.outputs.fdiVersions }}
+            cdiVersions: ${{ steps.versions.outputs.cdiVersions }}
+            nodeVersions: '["20"]'
+        steps:
+            - uses: actions/checkout@v4
+            - uses: supertokens/get-supported-versions-action@main
+              id: versions
+              with:
+                  has-fdi: true
+                  has-cdi: true
+
+    test:
+        runs-on: ubuntu-latest
+        needs: define-versions
+        strategy:
+            fail-fast: false
+            matrix:
+                node-version: ${{ fromJSON(needs.define-versions.outputs.nodeVersions) }}
+                fdi-version: ${{ fromJSON(needs.define-versions.outputs.fdiVersions) }}
+
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  path: supertokens-node
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: ${{ matrix.node-version }}
+
+            - uses: supertokens/get-versions-action@main
+              id: versions
+              with:
+                  driver-name: python
+                  fdi-version: ${{ matrix.fdi-version }}
+              env:
+                  SUPERTOKENS_API_KEY: ${{ secrets.SUPERTOKENS_API_KEY }}
+
+            - name: Start core
+              working-directory: supertokens-node
+              run: docker compose up --wait
+
+            - name: Install SDK dependencies
+              working-directory: supertokens-node
+              run: npm i
+
+            - name: Start test server
+              working-directory: supertokens-node/test/frontendIntegration
+              env:
+                  TEST_MODE: testing
+              run: |
+                  npm i
+
+                  NODE_PORT=8080 node . &> app-server.log &
+                  NODE_PORT=8082 node . &> app-server-cross-domain.log &
```

Comment:
if there is an easy way to do it we should wait for these to start up.

Reply:
and fail early if they didn't

Reply:
Sure, this can be done
Will make the change in all the workflows

Reply:
Added this to the website-testing-action, will carry over to all the workflows.
https://github.com/supertokens/website-testing-action/blob/main/action.yml#L96-L121